common/gpu/intel/default.nix

```
@@ -0,0 +1,15 @@
+{ config, lib, pkgs, ... }:
+
+{
+  boot.initrd.kernelModules = [ "i915" ];
+
+  environment.variables = {
+    VDPAU_DRIVER = lib.mkIf config.hardware.opengl.enable (lib.mkDefault "va_gl");
+  };
+
+  hardware.opengl.extraPackages = with pkgs; [
+    vaapiIntel
+    libvdpau-va-gl
+    intel-media-driver
+  ];
```

```suggestion
  hardware.opengl.extraPackages = with pkgs; [
    intel-media-driver
    intel-ocl
    vaapiIntel
  ];
```

see https://search.nixos.org/options?channel=unstable&show=hardware.opengl.extraPackages&from=0&size=50&sort=relevance&type=packages&query=hardware.opengl.extraPackages

---

`intel-ocl` would be a part of the intel cpu module, not the gpu. In addition it does not replace ` libvdpau-va-gl` which implements vdpau on top of `libva` and `libGL`.

---

So this means that there are intel CPUs without GPU that have opencl support?

---

Technically yes[^1], but it appears to be part of an SDK and is supposed to be linked statically into applications.

It seems that `intel-ocl` is GPU-specific and could probably be added to the existing intel module, but this PR does not change the behavior of existing modules so that should probably be a separate PR.

[^1]: https://www.intel.com/content/www/us/en/developer/articles/tool/opencl-drivers.html#cpu-section

---

fair enough.